web3_proxy/src/app/mod.rs

```diff
@@ -74,6 +76,46 @@ pub static APP_USER_AGENT: &str = concat!(
 // aggregate across 1 week
 pub const BILLING_PERIOD_SECONDS: i64 = 60 * 60 * 24 * 7;
 
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub enum UserTier {
+    #[default]
+    Free,
+    PrivateDemo,
+    EffectivelyUnlimited,
+    Unlimited,
+    PremiumOutOfFunds,
+    Premium,
+}
+
+impl ToString for UserTier {
+    fn to_string(&self) -> String {
+        match self {
+            UserTier::Free => "Free".to_owned(),
+            UserTier::PrivateDemo => "PrivateDemo".to_owned(),
+            UserTier::EffectivelyUnlimited => "EffectivelyUnlimited".to_owned(),
+            UserTier::Unlimited => "Unlimited".to_owned(),
+            UserTier::PremiumOutOfFunds => "PremiumOutOfFunds".to_owned(),
+            UserTier::Premium => "Premium".to_owned(),
+        }
+    }
+}
+
+impl TryFrom<&str> for UserTier {
```

Comment:
Why do we need this? This seems incredibly fragile to me. If we wanted user tiers in code like this, we should just remove them entirely from the database. 

There are going to be a bunch of tiers created for specific users. We don't want to have to coordinate deploys with tier changes. 

This needs to work differently

Reply:
I removed this, there are snippets like `if user_tier.title.as_str() != "Premium" {` for which I thought it would be better to have a struct. I changed it back now, let me know if that's better.